File: Views/Main/FavoritesCardView.swift
import SwiftUI

struct FavoritesCardView: View {
    @Environment(\.dismiss) private var dismiss
    @EnvironmentObject private var favoritesService: FavoritesService
    @EnvironmentObject private var appState: AppState
    
    @State private var currentIndex = 0
    @State private var dragOffset = CGSize.zero
    @State private var showingListView = false
    @State private var showingDeleteAlert = false
    
    private var currentFavoriteIndex: Int {
        return currentIndex
    }
    
    // Favoris visibles (3 maximum pour la performance)
    private var visibleFavorites: [(Int, FavoriteQuestion)] {
        let allFavorites = favoritesService.getAllFavorites()
        guard !allFavorites.isEmpty else { return [] }
        
        let startIndex = max(0, currentIndex - 1)
        let endIndex = min(allFavorites.count - 1, currentIndex + 1)
        
        var result: [(Int, FavoriteQuestion)] = []
        for i in startIndex...endIndex {
            result.append((i, allFavorites[i]))
        }
        return result
    }
    
    var body: some View {
        ZStack {
            // Fond gris clair identique aux autres pages
            Color(red: 0.97, green: 0.97, blue: 0.98)
                .ignoresSafeArea()
            
            VStack(spacing: 0) {
                // Header avec navigation
                HStack {
                    Spacer()
                    
                    // Titre
                    VStack(spacing: 4) {
                        Text("Favoris")
                            .font(.system(size: 28, weight: .bold))
                        .foregroundColor(.black)
                    }
                    
                    Spacer()
                }
                .padding(.horizontal, 20)
                .padding(.top, 20)
                .padding(.bottom, 20)
                
                // Contenu principal
                if favoritesService.getAllFavorites().isEmpty {
                    // État vide avec layout identique au journal
                    VStack(spacing: 30) {
                        Image(LocalizationService.localizedImageName(frenchImage: "mili", defaultImage: "manon"))
                            .resizable()
                            .aspectRatio(contentMode: .fit)
                            .frame(width: 240, height: 240)
                        
                        VStack(spacing: 12) {
                            Text("add_favorite_questions".localized)
                                .font(.system(size: 22, weight: .medium))
                                .foregroundColor(.black)
                                .multilineTextAlignment(.center)
                            
                            Text("add_favorites_description".localized)
                                .font(.system(size: 16))
                                .foregroundColor(.black.opacity(0.7))
                                .multilineTextAlignment(.center)
                                .padding(.horizontal, 30)
                        }
                    }
                    .frame(maxWidth: .infinity, maxHeight: .infinity)
                    .padding(.horizontal, 20)
                } else {
                    // Cartes de favoris avec le même design
                    GeometryReader { geometry in
                        let cardWidth = geometry.size.width - 40
                        let cardSpacing: CGFloat = 30
                        
                        ZStack {
                            ForEach(visibleFavorites, id: \.0) { indexAndFavorite in
                                let (index, favorite) = indexAndFavorite
                                let offset = CGFloat(index - currentFavoriteIndex)
                                let xPosition = offset * (cardWidth + cardSpacing) + dragOffset.width
                                
                                FavoriteQuestionCardView(
                                    favorite: favorite,
                                    isBackground: index != currentFavoriteIndex
                                )
                                .frame(width: cardWidth)
                                .offset(x: xPosition)
                                .scaleEffect(index == currentFavoriteIndex ? 1.0 : 0.95)
                                .opacity(index == currentFavoriteIndex ? 1.0 : 0.8)
                            }
                        }
                        .frame(width: geometry.size.width, height: geometry.size.height)
                        .position(x: geometry.size.width / 2, y: geometry.size.height / 2)
                        .gesture(
                            DragGesture()
                                .onChanged { value in
                                    dragOffset = value.translation
                                }
                                .onEnded { value in
                                    let threshold: CGFloat = 80
                                    let velocity = value.predictedEndTranslation.width - value.translation.width
                                    
                                    withAnimation(.spring(response: 0.6, dampingFraction: 0.8)) {
                                        if value.translation.width > threshold || velocity > 500 {
                                            // Swipe vers la droite - favori précédent
                                            if currentFavoriteIndex > 0 {
                                                currentIndex -= 1
                                            }
                                        } else if value.translation.width < -threshold || velocity < -500 {
                                            // Swipe vers la gauche - favori suivant
                                            if currentFavoriteIndex < favoritesService.getAllFavorites().count - 1 {
                                                currentIndex += 1
                                            }
                                        }
                                        
                                        // Remettre la carte en place
                                        dragOffset = .zero
                                    }
                                }
                        )
                    }
                    .frame(maxWidth: .infinity, maxHeight: .infinity)
                    .padding(.horizontal, 20)
                    
                    // Bouton Retirer des favoris (design moderne)
                    Button("remove_from_favorites".localized) {
                        showingDeleteAlert = true
                    }
                    .font(.system(size: 18, weight: .semibold))
                    .foregroundColor(.white)
                    .frame(width: UIScreen.main.bounds.width - 40) // Même largeur que les cartes
                    .frame(height: 56)
                    .background(
                        // Même couleur que le header des cartes
                        Color(red: 1.0, green: 0.4, blue: 0.6)
                    )
                    .cornerRadius(28)
                    .padding(.top, 20) // Réduit de 40 à 20 pour rapprocher du contenu
                    .padding(.bottom, 30) // Réduit de 50 à 30
                }
            }
        }
        .navigationBarHidden(true)
        .sheet(isPresented: $showingListView) {
            FavoritesView()
                .environmentObject(appState)
                .environmentObject(favoritesService)
        }
        .alert("remove_from_favorites".localized, isPresented: $showingDeleteAlert) {
            Button("cancel".localized, role: .cancel) { }
            Button("remove".localized, role: .destructive) {
                let allFavorites = favoritesService.getAllFavorites()
                if currentFavoriteIndex < allFavorites.count {
                    let currentFavorite = allFavorites[currentFavoriteIndex]
                    Task { @MainActor in
                        favoritesService.removeFavorite(questionId: currentFavorite.questionId)
                        
                        // Ajuster l'index si nécessaire
                        let updatedFavorites = favoritesService.getAllFavorites()
                        if currentIndex >= updatedFavorites.count && currentIndex > 0 {
                            currentIndex -= 1
                        }
                        
                        print("🔥 FavoritesCardView: Favori supprimé: \(currentFavorite.questionText.prefix(50))...")
                    }
                }
            }
        } message: {
            Text("remove_favorite_confirmation".localized)
        }
        .onAppear {
            print("🔥 FavoritesCardView: Vue des cartes favoris apparue")
            // Les favoris sont automatiquement chargés via le listener Firestore
        }
    }
}

struct FavoriteQuestionCardView: View {
    let favorite: FavoriteQuestion
    let isBackground: Bool
    
    var body: some View {
        VStack(spacing: 0) {
            // Header de la carte avec nom de catégorie (design moderne)
            VStack(spacing: 8) {
                Text(favorite.categoryTitle)
                    .font(.system(size: 18, weight: .bold))
                    .foregroundColor(.white)
                    .multilineTextAlignment(.center)
            }
            .frame(maxWidth: .infinity)
            .padding(.vertical, 20)
            .background(
                LinearGradient(
                    gradient: Gradient(colors: [
                        Color(red: 1.0, green: 0.4, blue: 0.6),
                        Color(red: 1.0, green: 0.6, blue: 0.8)
                    ]),
                    startPoint: .leading,
                    endPoint: .trailing
                )
            )
            
            // Corps de la carte avec la question favorite
            VStack(spacing: 30) {
                Spacer()
                
                Text(favorite.questionText)
                    .font(.system(size: 22, weight: .medium))
                    .foregroundColor(.white)
                    .multilineTextAlignment(.center)
                    .lineSpacing(6)
                    .padding(.horizontal, 30)
                
                Spacer()
                
                // Logo/Branding en bas (design moderne)
                HStack(spacing: 8) {
                    Image("leetchi2")
                        .resizable()
                        .aspectRatio(contentMode: .fit)
                        .frame(width: 24, height: 24)
                    
                    Text("Love2Love")
                        .font(.system(size: 16, weight: .semibold))
                        .foregroundColor(.white.opacity(0.9))
                }
                .padding(.bottom, 30)
            }
            .frame(maxWidth: .infinity, maxHeight: .infinity)
            .background(
                LinearGradient(
                    gradient: Gradient(colors: [
                        Color(red: 0.2, green: 0.1, blue: 0.15),
                        Color(red: 0.4, green: 0.2, blue: 0.3),
                        Color(red: 0.6, green: 0.3, blue: 0.2)
                    ]),
                    startPoint: .top,
                    endPoint: .bottom
                )
            )
        }
        .frame(maxWidth: .infinity)
        .frame(height: 400) // Réduit de 500 à 400 pour des cartes moins hautes
        .cornerRadius(20)
        .shadow(color: .black.opacity(isBackground ? 0.1 : 0.3), radius: isBackground ? 5 : 10, x: 0, y: isBackground ? 2 : 5)
    }
}

struct FavoritesCardView_Previews: PreviewProvider {
    static var previews: some View {
        FavoritesCardView()
            .environmentObject(FavoritesService())
    }
} 

File: Views/Main/FavoritesCardView.kt
package com.love2love.ui.favorites

import androidx.annotation.DrawableRes
import androidx.compose.foundation.Image
import androidx.compose.foundation.background
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.pager.HorizontalPager
import androidx.compose.foundation.pager.rememberPagerState
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material3.*
import androidx.compose.runtime.*
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.draw.shadow
import androidx.compose.ui.graphics.Brush
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.platform.LocalContext
import androidx.compose.ui.res.painterResource
import androidx.compose.ui.res.stringResource
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.style.TextAlign
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import androidx.compose.runtime.getValue
import androidx.compose.runtime.setValue
import androidx.compose.foundation.isSystemInDarkTheme
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.width
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.size
import androidx.compose.runtime.snapshots.SnapshotStateList
import androidx.compose.ui.layout.ContentScale
import androidx.compose.ui.tooling.preview.Preview
import androidx.compose.ui.text.style.TextOverflow
import androidx.compose.ui.text.font.FontStyle
import androidx.compose.ui.text.style.TextDecoration
import androidx.compose.ui.text.style.TextAlign.Companion.Center
import androidx.compose.ui.text.style.TextAlign.Companion.Start
import androidx.compose.ui.text.style.TextAlign.Companion.Justify
import androidx.compose.ui.text.style.TextAlign.Companion.End
import androidx.compose.ui.text.style.TextAlign.Companion.Left
import androidx.compose.ui.text.style.TextAlign.Companion.Right
import androidx.compose.foundation.*
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.foundation.layout.PaddingValues
import androidx.compose.foundation.pager.PagerDefaults
import androidx.compose.runtime.LaunchedEffect
import androidx.compose.runtime.mutableStateListOf
import androidx.compose.runtime.remember
import androidx.compose.runtime.rememberCoroutineScope
import kotlinx.coroutines.launch
import java.util.Locale

/**
 * Kotlin/Compose port of the SwiftUI FavoritesCardView.
 *
 * ✅ Uses Android standard localization via strings.xml with stringResource(...) inside @Composable.
 * ✅ Horizontal swipable cards using foundation HorizontalPager (Compose 1.4+).
 * ✅ Remove-from-favorites button with confirmation AlertDialog.
 * ✅ Empty state with locale-based image selection.
 *
 * Notes:
 * - Replace R.string.* and R.drawable.* with your actual resources.
 * - HorizontalPager is part of androidx.compose.foundation:pager. Ensure you have a recent Compose BOM/foundation.
 */

// ------------------------------------------------------------
// Data model
// ------------------------------------------------------------

data class FavoriteQuestion(
    val questionId: String,
    val questionText: String,
    val categoryTitle: String
)

// ------------------------------------------------------------
// Simple in-memory service (replace with your Firestore-backed service)
// ------------------------------------------------------------

object FavoritesService {
    private val _favorites: SnapshotStateList<FavoriteQuestion> = mutableStateListOf()
    fun getAllFavorites(): List<FavoriteQuestion> = _favorites
    fun setAllFavorites(items: List<FavoriteQuestion>) {
        _favorites.clear(); _favorites.addAll(items)
    }
    fun removeFavorite(questionId: String) {
        _favorites.removeAll { it.questionId == questionId }
    }
}

// ------------------------------------------------------------
// Screen
// ------------------------------------------------------------

@OptIn(ExperimentalFoundationApi::class)
@Composable
fun FavoritesCardScreen(
    modifier: Modifier = Modifier,
    // you can inject your real service via params or DI
    favoritesService: FavoritesService = FavoritesService
) {
    val context = LocalContext.current
    val favorites = favoritesService.getAllFavorites()

    var showDeleteDialog by remember { mutableStateOf(false) }
    var selectedIndexForDeletion by remember { mutableStateOf(0) }

    val pageCount = maxOf(favorites.size, 1)
    val pagerState = rememberPagerState(
        initialPage = 0,
        pageCount = { pageCount }
    )
    val scope = rememberCoroutineScope()

    // If the list shrinks (after deletion), keep pager in bounds
    LaunchedEffect(favorites.size) {
        val lastIndex = (favorites.size - 1).coerceAtLeast(0)
        if (pagerState.currentPage > lastIndex) {
            scope.launch { pagerState.scrollToPage(lastIndex) }
        }
    }

    // Background color identical to iOS page
    Box(
        modifier
            .fillMaxSize()
            .background(Color(red = 0.97f, green = 0.97f, blue = 0.98f))
    ) {
        Column(
            modifier = Modifier.fillMaxSize(),
            verticalArrangement = Arrangement.Top,
            horizontalAlignment = Alignment.CenterHorizontally
        ) {
            // Header
            Row(
                modifier = Modifier
                    .fillMaxWidth()
                    .padding(horizontal = 20.dp)
                    .padding(top = 20.dp, bottom = 20.dp),
                verticalAlignment = Alignment.CenterVertically,
                horizontalArrangement = Arrangement.Center
            ) {
                Text(
                    text = stringResource(id = R.string.favorites_title), // "Favoris"
                    fontSize = 28.sp,
                    fontWeight = FontWeight.Bold,
                    color = Color.Black
                )
            }

            // Content
            if (favorites.isEmpty()) {
                EmptyFavoritesState()
            } else {
                // Cards pager
                HorizontalPager(
                    state = pagerState,
                    beyondBoundsPageCount = 1,
                    pageSpacing = 30.dp,
                    contentPadding = PaddingValues(horizontal = 20.dp),
                    modifier = Modifier
                        .fillMaxWidth()
                        .weight(1f)
                ) { page ->
                    val favorite = favorites.getOrNull(page)
                    FavoriteQuestionCard(
                        favorite = favorite,
                        isBackground = page != pagerState.currentPage
                    )
                }

                // Remove button
                Button(
                    onClick = {
                        selectedIndexForDeletion = pagerState.currentPage
                        showDeleteDialog = true
                    },
                    modifier = Modifier
                        .fillMaxWidth()
                        .padding(horizontal = 20.dp)
                        .height(56.dp),
                    shape = RoundedCornerShape(28.dp),
                    colors = ButtonDefaults.buttonColors(
                        containerColor = Color(red = 1.0f, green = 0.4f, blue = 0.6f),
                        contentColor = Color.White
                    )
                ) {
                    Text(
                        text = stringResource(id = R.string.remove_from_favorites),
                        fontSize = 18.sp,
                        fontWeight = FontWeight.SemiBold
                    )
                }

                Spacer(Modifier.height(30.dp))
            }
        }

        if (showDeleteDialog) {
            AlertDialog(
                onDismissRequest = { showDeleteDialog = false },
                confirmButton = {
                    TextButton(onClick = {
                        val current = favorites.getOrNull(selectedIndexForDeletion)
                        if (current != null) {
                            favoritesService.removeFavorite(current.questionId)
                        }
                        showDeleteDialog = false
                    }) {
                        Text(text = stringResource(id = R.string.remove))
                    }
                },
                dismissButton = {
                    TextButton(onClick = { showDeleteDialog = false }) {
                        Text(text = stringResource(id = R.string.cancel))
                    }
                },
                title = { Text(text = stringResource(id = R.string.remove_from_favorites)) },
                text = { Text(text = stringResource(id = R.string.remove_favorite_confirmation)) }
            )
        }
    }
}

// ------------------------------------------------------------
// Empty state
// ------------------------------------------------------------

@Composable
private fun EmptyFavoritesState() {
    val context = LocalContext.current
    val imageRes = when (Locale.getDefault().language) {
        "fr" -> R.drawable.mili
        "de" -> R.drawable.crypto
        else -> R.drawable.manon
    }

    Column(
        modifier = Modifier
            .fillMaxWidth()
            .weight(1f)
            .padding(horizontal = 20.dp),
        verticalArrangement = Arrangement.Center,
        horizontalAlignment = Alignment.CenterHorizontally
    ) {
        Image(
            painter = painterResource(id = imageRes),
            contentDescription = null,
            modifier = Modifier
                .size(240.dp)
                .padding(bottom = 12.dp),
            contentScale = ContentScale.Fit
        )

        Text(
            text = stringResource(id = R.string.add_favorite_questions),
            fontSize = 22.sp,
            fontWeight = FontWeight.Medium,
            color = Color.Black,
            textAlign = TextAlign.Center,
            modifier = Modifier.padding(vertical = 4.dp)
        )
        Text(
            text = stringResource(id = R.string.add_favorites_description),
            fontSize = 16.sp,
            color = Color.Black.copy(alpha = 0.7f),
            textAlign = TextAlign.Center,
            modifier = Modifier.padding(horizontal = 30.dp)
        )
    }
}

// ------------------------------------------------------------
// Card
// ------------------------------------------------------------

@Composable
private fun FavoriteQuestionCard(
    favorite: FavoriteQuestion?,
    isBackground: Boolean
) {
    // Header gradient colors
    val headerGradient = Brush.linearGradient(
        colors = listOf(
            Color(red = 1.0f, green = 0.4f, blue = 0.6f),
            Color(red = 1.0f, green = 0.6f, blue = 0.8f)
        )
    )
    // Body gradient colors
    val bodyGradient = Brush.linearGradient(
        colors = listOf(
            Color(red = 0.2f, green = 0.1f, blue = 0.15f),
            Color(red = 0.4f, green = 0.2f, blue = 0.3f),
            Color(red = 0.6f, green = 0.3f, blue = 0.2f)
        )
    )

    Column(
        modifier = Modifier
            .fillMaxWidth()
            .height(400.dp)
            .shadow(if (isBackground) 5.dp else 10.dp, RoundedCornerShape(20.dp))
            .clip(RoundedCornerShape(20.dp))
    ) {
        // Header with category title
        Box(
            modifier = Modifier
                .fillMaxWidth()
                .background(headerGradient)
                .padding(vertical = 20.dp),
            contentAlignment = Alignment.Center
        ) {
            Text(
                text = favorite?.categoryTitle ?: "",
                fontSize = 18.sp,
                fontWeight = FontWeight.Bold,
                color = Color.White,
                textAlign = TextAlign.Center,
                modifier = Modifier.padding(horizontal = 16.dp)
            )
        }

        // Body with question + branding
        Column(
            modifier = Modifier
                .fillMaxSize()
                .background(bodyGradient),
            verticalArrangement = Arrangement.SpaceBetween,
            horizontalAlignment = Alignment.CenterHorizontally
        ) {
            Spacer(Modifier.height(12.dp))

            Text(
                text = favorite?.questionText ?: "",
                fontSize = 22.sp,
                fontWeight = FontWeight.Medium,
                color = Color.White,
                textAlign = TextAlign.Center,
                lineHeight = 28.sp,
                modifier = Modifier.padding(horizontal = 30.dp)
            )

            Column(
                modifier = Modifier
                    .fillMaxWidth()
                    .padding(bottom = 30.dp),
                horizontalAlignment = Alignment.CenterHorizontally
            ) {
                Row(verticalAlignment = Alignment.CenterVertically) {
                    Image(
                        painter = painterResource(id = R.drawable.leetchi2),
                        contentDescription = null,
                        modifier = Modifier.size(24.dp)
                    )
                    Spacer(Modifier.width(8.dp))
                    Text(
                        text = "Love2Love",
                        fontSize = 16.sp,
                        fontWeight = FontWeight.SemiBold,
                        color = Color.White.copy(alpha = 0.9f)
                    )
                }
            }
        }
    }
}

// ------------------------------------------------------------
// Preview (uses fake data)
// ------------------------------------------------------------

@Preview(showBackground = true)
@Composable
private fun FavoritesCardScreenPreview() {
    FavoritesService.setAllFavorites(
        listOf(
            FavoriteQuestion("1", "Quelle est la petite attention qui t’a le plus marqué(e) récemment ?", "Questions profondes"),
            FavoriteQuestion("2", "Quel souvenir commun te donne instantanément le sourire ?", "Souvenirs"),
            FavoriteQuestion("3", "Qu’aimerais-tu que l’on crée ensemble ce mois-ci ?", "Projets")
        )
    )
    FavoritesCardScreen()
}

/* ------------------------------------------------------------
   strings.xml — add these keys (FR shown as example)
------------------------------------------------------------

<resources>
    <string name="favorites_title">Favoris</string>
    <string name="add_favorite_questions">Ajoute tes questions préférées</string>
    <string name="add_favorites_description">Marque des questions en favori pour les retrouver facilement ici et revenir dessus quand vous voulez.</string>
    <string name="remove_from_favorites">Retirer des favoris</string>
    <string name="remove">Retirer</string>
    <string name="cancel">Annuler</string>
    <string name="remove_favorite_confirmation">Es-tu sûr de vouloir retirer cette question de tes favoris ?</string>
</resources>

Notes:
- Utilise des répertoires spécifiques à la locale pour les images si tu veux automatiser la sélection (ex: res/drawable-fr/mili.png, res/drawable-de/crypto.png, res/drawable/manon.png en défaut). Le when(Locale…) ci‑dessus est un fallback si tu préfères gérer par code.
- Remplace R.drawable.leetchi2 par ton icône.
- Si tu préfères Context.getString(...) hors d’un @Composable, utilise LocalContext.current.getString(R.string.key).
*/
